js/modules/tamagotchi.js

```diff
@@ -85,12 +110,17 @@ export default class Tamagotchi {
 
 		const currentState = states.sort((a, b) => b.importance - a.importance).find(state => state.value <= 6);
 
-		if (this.health.value >= 7 && this.hunger.value >= 7 && this.fun.value >= 7 && this.energy.value >= 7) {
-			tamagoSpirit.classList.add('happy');
-			tamagoText.textContent = tamagoSpirit.classList[1].toUpperCase();
-		} else if (currentState) {
```

Comment:
**issue:** To prevent spaghetti you should operate on Tamagotchi `status`, eg:

```ts

checkCurrentStatus() {
   switch (this.status) {
       case "hungry":
       tamagoSpirit.classList.add('hungry');
       break;
       case "eating":
       tamagoSpirit.classList.add('eating');
       break;
       case "happy":
       tamagoSpirit.classList.add('happy');
       break;
   }
}

```